src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,1323 @@
+
+        :return calibration_data (CalibrationLoopParameters object): Object holding the calibration data which can be
+        send to the ASM API
+        """
+        descanner = self._mirror_descanner
+        scanner = self._ebeam_scanner
+        mppc = self._mppc
+
+        descan_rotation = descanner.rotation.value
+        descan_offset = descanner.scanOffset.value
+        dwell_time = scanner.dwellTime.value
+        dwell_time_ticks = scanner.getTicksDwellTime()
+        scan_rotation = scanner.rotation.value
+        scan_delay = scanner.getTicksScanDelay()
+        scan_offset = scanner.scanOffset.value
+        resolution = mppc.cellCompleteResolution.value[0]
+
+        # Check if the descanner clockperiod is still a multiple of de scanner clock period, otherwise raise error.
+        if not almost_equal(descanner.clockPeriod.value % scanner.clockPeriod.value, 0):
+            logging.error("Descanner and/or scanner clock period changed which means the descanner period is no "
+                          "longer a whole multiple of scanner clock periods. This means that calibration is no "
+                          "longer accurate.")
+            raise ValueError("Descanner clock period is no longer a whole multiple of the scanner clock period.")
+
+        # Creation of setpoints
+        flyback_time = descanner.physicalFlybackTime
+        remainder_scanning_time = (dwell_time * resolution) % descanner.clockPeriod.value
+        if remainder_scanning_time is not 0:
+            # Add adjusted flyback time if there is a remainder of scanning time by adding one setpoint to ensure the
+            # total scanning period is equal to a whole number of descan clock periods.
+            flyback_time = flyback_time + descanner.clockPeriod.value
+
+        # Period of the calibration signal in seconds
+        calibration_period = (dwell_time * resolution) + flyback_time
+
+        # Support a calibration freq from 125 to 25000 Hz
+        if not scanner.dwellTime.range[0] < calibration_period < 0.008:
+            logging.error("Cannot perform calibration using a calibration frequency lower than 125 or 5000. With the "
+                          "given parameters the required calibration frequency would be %s." % (1/calibration_period))
+            raise ValueError("Calibration of given values requires a calibration frequency which is out of range.")
+        calibration_frequency = 1/calibration_period
+
+        # One period with points at a sampling period of the descanner clock period
+        time_points_descanner = numpy.arange(0, calibration_period, descanner.clockPeriod.value)
+        x_descan_setpoints = descanner.scanGain.value[0] * numpy.sin(2 * math.pi * calibration_frequency * time_points_descanner)
+        y_descan_setpoints = descanner.scanGain.value[1] * signal.sawtooth(2 * math.pi * calibration_frequency *
```

Comment:
Currently the setter sets the gain values to a default value of 0. If for some reason the user like to calibrate the system using values where the scanner y values have a gain of 0. (and so all the descanner values a gain of 1) than this would still be an option this way.
Calculating all the y_descan_setpoints may therefore seem like a unnecessary part to do, but since this takes almost no time it only makes sense to remove it for code clarity.
Summarizing: the descanner.scanGain is set by default to 0, if a user likes to also tweaks something differently now it is still possible. But changes are that that won't be used often.

Reply:
Yes then leave it like it is for now.
_"Calculating all the y_descan_setpoints may therefore seem like a unnecessary part to do, but since this takes almost no time it only makes sense to remove it for code clarity."_ -> It is definitely not unnecessary. On contrary, we explicitly have to send these zeros.